### PR TITLE
support kube-apiserver pool

### PR DESCRIPTION
**Pull request by l337ch**

### CODE REVIEW

**Comment by l337ch** [comment]
Not ready to merge yet. Forgot to add sample yaml.

**Comment by l337ch** [comment]
also, the apiserver is using the node attributes on aws.

File: kubernetes/Vagrantfile [hunk 1]
```diff
@@ -130,6 +130,27 @@ def final_node_ip
   get_vm_ip(get_num_vms)
 end
 
+def get_num_apiservers
+  settings = cluster_settings
+  settings['cluster']['apiservers'] || 1
+end
+
+def get_total_instances
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: I think this is the same as `get_num_vms` ?

File: kubernetes/Vagrantfile [hunk 2]
```diff
@@ -130,6 +130,27 @@ def final_node_ip
...
+  # Add 2 for etcd and master servers
+  get_num_nodes.to_i + get_num_apiservers.to_i + 2
+end
+
+def apiserver_pool
+  pool = []
+  tot_wout_apiservers = get_num_nodes + 3
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: this seems weird, because `get_num_nodes` already has the number of api servers summed into it, what's this extra 3?

Comment (spiffxp) [inline kubernetes/Vagrantfile]: What's the 3 here?

File: kubernetes/Vagrantfile [hunk 3]
```diff
@@ -141,6 +162,11 @@ end
 def get_coreos_node_info(host_number)
   node_info = Hash.new
   settings = cluster_settings
+  # SubTotal of instances excluding apiservers
+  subtotal = get_num_nodes + 2
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: `get_num_vms - get_num_apiservers` would express the comment better

File: kubernetes/Vagrantfile [hunk 4]
```diff
@@ -141,6 +162,11 @@ end
...
+  # Total number of instances including apiservers
+  total = subtotal + get_num_apiservers
```
Comment (spiffxp) [inline kubernetes/Vagrantfile]: Is this the same as `get_num_vms`?

File: kubernetes/Vagrantfile (hunk 1 above)
Comment (l337ch) [inline kubernetes/Vagrantfile]: right, i thought i deleted that after your PR.

File: kubernetes/Vagrantfile (hunk 2 above)
Comment (l337ch) [inline kubernetes/Vagrantfile]: To start obtaining the IP for the apiserver(s) which the next one after the final node(+1)
2(etcd, master) + 1

Comment (l337ch) [inline kubernetes/Vagrantfile]: Hmm, is that math right. really ugly way to do this

File: kubernetes/Vagrantfile (hunk 4 above)
Comment (l337ch) [inline kubernetes/Vagrantfile]: yup, modded and commited

File: kubernetes/Vagrantfile (hunk 3 above)
Comment (l337ch) [inline kubernetes/Vagrantfile]: i didn't really touch much after your commits :)

**Comment by l337ch** [comment]
ok this should be good to go

**Comment by spiffxp** [comment]
LGTM

**Pull request merged**